logging/LoggingService.cpp

```diff
@@ -21,11 +21,35 @@ LoggingService::LoggingService(std::string name) :
         levels_prop("Levels","A PropertyBag defining the level of each category of interest."),
         additivity_prop("Additivity","A PropertyBag defining the additivity of each category of interest."),
         appenders_prop("Appenders","A PropertyBag defining the appenders for each category of interest."),
+        level_EMERG_attr ("EMERG",  log4cpp::Priority::EMERG ),
+        level_FATAL_attr ("FATAL",  log4cpp::Priority::FATAL ),
+        level_ALERT_attr ("ALERT",  log4cpp::Priority::ALERT ),
+        level_CRIT_attr  ("CRIT",   log4cpp::Priority::CRIT  ),
+        level_ERROR_attr ("ERROR",  log4cpp::Priority::ERROR ),
+        level_WARN_attr  ("WARN",   log4cpp::Priority::WARN  ),
+        level_NOTICE_attr("NOTICE", log4cpp::Priority::NOTICE),
+        level_INFO_attr  ("INFO",   log4cpp::Priority::INFO  ),
+        level_DEBUG_attr ("DEBUG",  log4cpp::Priority::DEBUG ),
+        level_NOTSET_attr("NOTSET", log4cpp::Priority::NOTSET),
+        setCategoryPriority_mtd("setCategoryPriority", &LoggingService::setCategoryPriority, this),
+        getCategoryPriorityName_mtd("getCategoryPriorityName", &LoggingService::getCategoryPriorityName, this),
         logCategories_mtd("logCategories", &LoggingService::logCategories, this)
 {
     this->properties()->addProperty( levels_prop );
     this->properties()->addProperty( additivity_prop );
     this->properties()->addProperty( appenders_prop );
+    this->provides()->addAttribute(level_EMERG_attr);
+    this->provides()->addAttribute(level_FATAL_attr);
+    this->provides()->addAttribute(level_ALERT_attr);
+    this->provides()->addAttribute(level_CRIT_attr);
+    this->provides()->addAttribute(level_ERROR_attr);
+    this->provides()->addAttribute(level_WARN_attr);
+    this->provides()->addAttribute(level_NOTICE_attr);
+    this->provides()->addAttribute(level_INFO_attr);
+    this->provides()->addAttribute(level_DEBUG_attr);
+    this->provides()->addAttribute(level_NOTSET_attr);
```

Comment:
Use `RTT::Constant` instead? You can add a new constant without the need to add them as member variables with
```cpp
this->provides()->setValue(new Constant<int>("EMERG", log4cpp::Priority::EMERG));
// ...
```
[addConstant()](https://github.com/orocos-toolchain/rtt/blob/master/rtt/ConfigurationInterface.hpp#L135) cannot be used here because it requires a const reference and can lead to linker errors or undefined results if you would pass an enum value.